src/isomorphic/classic/class/ReactClass.js

```diff
@@ -437,6 +437,21 @@ function validateMethodOverride(isAlreadyDefined, name) {
  */
 function mixSpecIntoComponent(Constructor, spec) {
   if (!spec) {
+    if (__DEV__) {
+      var typeofSpec = typeof spec;
+      var isMixinValid = typeofSpec === 'object' && spec !== null;
+
+      warning(
+        isMixinValid,
+        '%s: You\'re attempting to include a mixin that is either null ' +
+        'or not an object. Check the mixins included by the component, ' +
+        'as well as any mixins they include themselves. ' +
+        'Expected object but got %s.',
```

Comment:
I mean that we should avoid confusing warnings even if turn up in edge cases.
We could do something like

```js
var type;
if (spec == null) {
  type = '' + spec;
} else if (Array.isArray(spec)) {
  type = 'array';
} else {
  type = typeof spec;
}
```